packages/mobile/src/tokens/saga.ts

```diff
@@ -11,31 +11,71 @@ import { addStandbyTransaction, removeStandbyTransaction } from 'src/transaction
 import { TransactionStatus, TransactionTypes } from 'src/transactions/reducer'
 import { sendAndMonitorTransaction } from 'src/transactions/saga'
 import Logger from 'src/utils/Logger'
-import { web3 } from 'src/web3/contracts'
+import { contractKit, web3 } from 'src/web3/contracts'
 import { getConnectedAccount, getConnectedUnlockedAccount } from 'src/web3/saga'
 import * as utf8 from 'utf8'
 
 const TAG = 'tokens/saga'
 
+// The number of wei that represent one unit in a contract
+const contractWeiPerUnit: { [key in CURRENCY_ENUM]: BigNumber | null } = {
```

Comment:
Can a wei per unit differ for two tokens? If not probably not need to cache two diff values for it

Reply:
Yes it can differ and is set different for gold and stabletokens, it's a configurable parameter here: https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L113